src/Window.cpp

```diff
@@ -146,52 +145,50 @@ void Window::handle_launch()
 
 void Window::handle_login()
 {
-  this->_loginWidgetUI.btn_login->setDisabled(true); // probably a valid mutex?
+  this->_loginWidgetUI.btn_login->setDisabled(true); // lock mutex (probably safe?)
 
   auto username = this->_loginWidgetUI.input_username->text().toStdString();
```

Comment:
"Everything can be const... unless it can't"
Make sure every variable is const.